### PR TITLE
Pinot 1.1.0

### PR DESCRIPTION
**Pull request by elonazoulay**
## Description

Upgrade Pinot to use latest libraries. 

I did not have permissions to update #21234 branch so I created a new pr.
This is a continuation of this pr (now closed since pinot released 1.1.0) #19136

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-pinot/pom.xml [hunk 1]
```diff
@@ -14,15 +14,32 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <dep.pinot.version>0.12.1</dep.pinot.version>
+        <dep.pinot.version>1.1.0</dep.pinot.version>
+        <!--
+           Project's default for air.test.parallel is 'methods'. By design, 'instances' makes TestNG run tests from one class in a single thread.
```
Comment (ebyhr) [inline plugin/trino-pinot/pom.xml:19]: I think Pinot connector already migrated to JUnit from TestNG. https://github.com/trinodb/trino/commit/8025f2e9f12226b146a66e6ec3c21a12a6c2d254

File: plugin/trino-pinot/pom.xml [hunk 2]
```diff
@@ -14,15 +14,32 @@
...
+           As a side effect, it prevents TestNG from initializing multiple test instances upfront, which happens with 'methods'.
+           A potential downside can be long tail single-threaded execution of a single long test class.
+           TODO (https://github.com/trinodb/trino/issues/11294) remove when we upgrade to surefire with https://issues.apache.org/jira/browse/SUREFIRE-1967
+           -->
+        <air.test.parallel>instances</air.test.parallel>
+        <air.test.jvm.additional-arguments>${air.test.jvm.additional-arguments.default}
```
Comment (ebyhr) [inline plugin/trino-pinot/pom.xml:25]: Are these arguments required not only in tests but also in actual Trino clusters?

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java [hunk 3]
```diff
@@ -38,4 +38,27 @@ protected boolean isGrpcEnabled()
...
+        abort("Broken :( Elon help!");
+    }
+
+    @Test
+    @Override
+    public void testTransformFunctions()
+    {
+        abort("Broken :( Elon help!");
+    }
+     */
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java:63]: Let's remove obsolete comments.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java [hunk 4]
```diff
@@ -65,8 +65,8 @@
 public class TestingPinotCluster
         implements Closeable
 {
-    public static final String PINOT_LATEST_IMAGE_NAME = "apachepinot/pinot:0.12.1";
-    public static final String PINOT_PREVIOUS_IMAGE_NAME = "apachepinot/pinot:0.11.0";
+    public static final String PINOT_LATEST_IMAGE_NAME = "apachepinot/pinot:1.1.0";
+    public static final String PINOT_PREVIOUS_IMAGE_NAME = "apachepinot/pinot:0.12.1";
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java:69]: Could you update `Requirements` in pinot.md?

File: plugin/trino-pinot/pom.xml (hunk 1 above)
Comment (elonazoulay) [reply plugin/trino-pinot/pom.xml:19]: Thanks!

File: plugin/trino-pinot/pom.xml (hunk 2 above)
Comment (elonazoulay) [reply plugin/trino-pinot/pom.xml:25]: Yes

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java (hunk 3 above)
Comment (elonazoulay) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java:63]: good catch, will do:)

**Comment by wendigo** [comment]
Let's move discussion back to the original PR: https://github.com/trinodb/trino/pull/21234. I've applied your changes @elonazoulay, added co-authorship and cleaned up a bit.

**Comment by wendigo** [comment]
Merged into master.

**Pull request closed without merging**